packages/nats-runner/src/NatsRunner.ts

```diff
@@ -167,32 +198,40 @@ export class NatsRunner {
         this.logger.debug(`Registering airlock handler for ${subject}`);
         const subscriptionOptions = handler.getSubscriptionOptions();
 
-        const subscription = this.natsConnection.subscribe(subject, subscriptionOptions);
+        const subscription = this.natsConnection.subscribe(
+            subject,
+            subscriptionOptions
+        );
 
         this.handleAirlockMessage(subscription, handler);
     }
 
-    async handleAirlockMessage(subscription: Subscription, handler: AirlockHandler): Promise<void> {
+    async handleAirlockMessage(
+        subscription: Subscription,
+        handler: AirlockHandler
+    ): Promise<void> {
         for await (const message of subscription) {
             try {
-                const response = await handler.handle(new AirlockMessage(message));
+                const response = await handler.handle(
+                    new AirlockMessage(message)
+                );
 
                 message.respond(this.jsonCodec.encode(response));
             } catch (err) {
                 this.logger.error((err as Error).message);
                 message.respond(
                     this.jsonCodec.encode({
-                        error:
-                            err instanceof JWAError
-                                ? JSON.stringify({ ...err, origin: undefined })
-                                : (err as Error).message
+                        error: (err as Error).message
```

Comment:
The JWAError serialization is missing.

Reply:
not familiar with that, could you give more details?

Reply:
Yes, the error management has been refactored and every error thrown by our services is now a JWAError. The origin is dismissed because Airlock doesn't need it.
You can find how it works here: [Notion - JWAError](https://www.notion.so/playtix/Error-handling-in-services-7a5d769c715042009a3d1fd93027ccf1)